monitored-httpclient/src/main/java/com/github/nhenneaux/resilienthttpclient/monitoredclientpool/ResilientClient.java

```diff
@@ -104,9 +104,12 @@ private static <T> CompletableFuture<HttpResponse<T>> handleConnectTimeout(
             httpResponseCompletableFuture.completeExceptionally(new HttpConnectTimeoutException("Cannot connect to the server, the following address were tried without success " + triedAddress + "."));
             return httpResponseCompletableFuture;
         }
-        return Optional.of(firstClient)
+
+        final Optional<SingleIpHttpClient> currentSingleIpHttpClient = Optional.of(firstClient)
                 .filter(ignored -> triedAddress.isEmpty())
-                .or(roundRobinPool::next)
+                .or(roundRobinPool::next);
+
```

Comment:
can be reverted since no change or I missed it?

Reply:
Change is here actually:

[currentSingleIpHttpClient.ifPresent(SingleIpHttpClient::refreshCounterWithException);](https://github.com/nhenneaux/resilient-httpclient/pull/50/files/54aa8d5076d5eb396efc7b3b68dc9649d663f9a5#diff-63b5f124475a681d4708e3e27fabc662176086950c2c985413f7d66d26be104aR126)

and [here](https://github.com/nhenneaux/resilient-httpclient/pull/50/files/54aa8d5076d5eb396efc7b3b68dc9649d663f9a5#diff-63b5f124475a681d4708e3e27fabc662176086950c2c985413f7d66d26be104aR136)

Reply:
so this line change can be removed ?

Reply:
It should stay. We need to use this variable to update count whether exceptionally or normal